src/System.ServiceModel.Primitives/tests/Description/OperationContractAttributeTest.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Runtime.Serialization;
+using System.ServiceModel;
+using System.ServiceModel.Description;
+using System.Threading.Tasks;
+using Infrastructure.Common;
+using Xunit;
+
+public static class OperationContractAttributeTest
+{
+    [WcfFact]
+    public static void Default_Ctor_Initializes_Correctly()
+    {
+        OperationContractAttribute oca = new OperationContractAttribute();
+
+        Assert.True(oca.IsInitiating, "IsInitiating should have been true");
+        Assert.True(!oca.IsTerminating, "IsTerminating should have been false");
+    }
+
+    [WcfFact]
+    public static void IsInitiating_Method_Requires_Proper_SessionMode()
+    {
+        var channelFactory = CreateChannelFactoryHelper<INonInitiatingNonTerminatingService>("net.tcp://localhost/dummy.svc");
+        // The default ServiceContract's SessionMode.Allowed is not compatible with an OperationContract that has IsInitiating or IsTerminating
+        var exception = Assert.Throws<InvalidOperationException>(() =>
+        {
+            channelFactory.CreateChannel();
+        });
+        Assert.True(exception.Message.Contains("The contract 'INonInitiatingNonTerminatingService' is not self-consistent"));
+    }
```

Comment:
This test should fail on systems running in a different locale if/when localization is working. Just check for the existence of the interface name in the message.